index.js

```diff
@@ -143,18 +156,17 @@ const parseArgs = (
           StringPrototypeSlice(arg, 0, index),
           StringPrototypeSlice(arg, index + 1),
           result);
-      } else if (pos + 1 < argv.length &&
-        !StringPrototypeStartsWith(argv[pos + 1], '-')
-      ) {
+      } else if (peek(argv, pos) !== TOKENS.FLAG) {
         // withValue option should also support setting values when '=
         // isn't used ie. both --foo=b and --foo b should work
 
-        // If withValue option is specified, take next position arguement as
+        // If withValue option is specified, take next position argument as
         // value and then increment pos so that we don't re-evaluate that
         // arg, else set value as undefined ie. --foo b --bar c, after setting
         // b as the value for foo, evaluate --bar next and skip 'b'
         const val = options.withValue &&
-          ArrayPrototypeIncludes(options.withValue, arg) ? argv[++pos] :
+          (ArrayPrototypeIncludes(options.withValue, arg) ||
+           ArrayPrototypeIncludes(options.multiples, arg)) ? argv[++pos] :
```

Comment:
Why have you added a test for `multiples` here? That does not look right for a multiple flag. The logic for multiples is in `storeOptionValue`.

Reply:
The behavior of multiples mixing positional and '='s appeared to be buggy, I can add a test for this separately.